tensilelite/rocisa/rocisa/include/helper.hpp

```diff
@@ -47,50 +49,87 @@ inline std::string getGfxNameTuple(const IsaVersion& isaVersion)
            + int_to_hex[isaVersion[2]];
 }
 
-inline std::pair<int, std::string>
-    run(const std::vector<char*>& cmd, const std::string& input, bool debug = false)
+#define CHECK_COMGR(call)                                                                  \
```

Comment:
Tip; you should almost never need a macro like this defined in a header for a local use case. This is a sure sign that you should be using a real translation unit.

Reply:
Further nit: since this is just raising an exception on failure, it is perfect for an inline static function, not a macro. Only use but multi line macros like this if doing something structural (ie. Return on error, etc). If just wanting to capture a preprocessor value like the current location, have a one line check macro that calls the check function, with most of the logic in a real function.